### PR TITLE
Buttons: enable focus shadow for borderless buttons

### PR DESCRIPTION
**Pull request by drw158**
This give move visibility to buttons while using the keyboard to
navigate. Also enabling border radius for style consistency.

Before/After

screen shot 2017-07-12 at 4 39 41 pm
screen shot 2017-07-12 at 4 39 25 pm

## Question

The shadow is showing up when I click on `scary` borderless buttons but not on regular borderless buttons. Why? We don't want the shadow to show up while clicking. 

Happens in Chrome only 

jul-12-2017 16-43-10

Also happens on regular big buttons, so it's an existing issue

jul-12-2017 16-46-50

### CODE REVIEW

**Comment by folletto** [comment]
I think the "scary" ones is a bug. Here's why:

1. On mouse click, we don't show the focus ring, as the button gets highlighted and then there should be an action. It also already shows an animation.
2. On focus instead (tab/keyboard navigation) we do show the focus ring, as that's the cursor that allow people navigate the UI.

If you check both standard (white) and primary (blue) buttons work in this way. 

There seem to be some bug for Scary buttons as instead that doesn't work. I haven't checked the internals yet, but as far as I know the difference is in the root tag:

* `.accessible-focus *:focus` — shows focus only on keyboard nav
* `*:focus` — shows focus even on mouse click

So I guess we should fix all the button focus styles to apply only on `.accessible-focus`.

**Comment by drw158** [comment]
Oh! I did not know about that class, thank you. I will investigate.

**Comment by drw158** [comment]
`scary` is unique because it needs to override the blue shadow with a red one. It seems the `.accessible-focus` class was overlooked while doing this. 

I just nested the red shadow override inside the `.accessible-focus` class and it seems to work:

```	scss
.accessible-focus & {
		&:focus {
			box-shadow: 0 0 0 2px lighten( $alert-red, 20% );
		}
	}
```
jul-13-2017 16-18-49

**Comment by folletto** [comment]
Tested in Chrome, works 👍

File: client/components/button/style.scss [hunk 1]
```diff
@@ -143,9 +143,13 @@ button {
 	&:focus {
 		border-color: $alert-red;
 	}
-	&:focus {
-		box-shadow: 0 0 0 2px lighten( $alert-red, 20% );
+
+	.accessible-focus & {
+		&:focus {
```
Comment (folletto) [inline client/components/button/style.scss]: Any reason for indenting this instead of a single line?

`.accessible-focus &:focus {`

Comment (drw158) [inline client/components/button/style.scss]: No reason other than just copy/pasting the same code found at the top. I think yours makes sense.

Comment (drw158) [inline client/components/button/style.scss]: @folletto Your method compiles the same, so I updated the code in both places. Thanks!

**Comment by folletto** [comment]
Let's 🚢  this then!

**Comment by blowery** [comment]
@folletto @drw158 this seems to have broken the build on master..

```
{
  "formatted": "Error: Invalid CSS after \"\t}\": expected selector or at-rule, was \"}\"\n        on line 107 of client/components/button/style.scss\n>> \t}\n   --^\n",
  "message": "Invalid CSS after \"\t}\": expected selector or at-rule, was \"}\"",
  "column": 3,
  "line": 107,
  "file": "client/components/button/style.scss",
  "status": 1
}
{
  "formatted": "Error: Invalid CSS after \"\t}\": expected selector or at-rule, was \"}\"\n        on line 107 of client/components/button/style.scss\n>> \t}\n   --^\n",
  "message": "Invalid CSS after \"\t}\": expected selector or at-rule, was \"}\"",
  "column": 3,
  "line": 107,
  "file": "client/components/button/style.scss",
  "status": 1
}
```

**Comment by blowery** [comment]
reverted in #16251

**Pull request merged**